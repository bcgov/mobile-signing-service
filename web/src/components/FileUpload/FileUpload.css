.file-upload-container {
  background-color: white;
  /* margin: 0em 3em 0em 3em; */
}

.file-upload-container .drop-zone-content {
  text-align: center;
  height: 4em;
  /* width: 16em; */
  background-color: white;
  border-style: dashed;
  border-width: 1px;
  border-color: lightgray;
  border-radius: 0.5em;
  color: darkgrey;
  /* display: inline-block; */
  padding: 0.5em 0.5em 0;
}

.file-icon {
  color: #fcba19;
  margin: 0 0.3em 0 0;
}

/* .file-upload-container:disabled {
  background-color: lightcoral;
} */

.file-upload-container ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  text-align: left;
}

.drop-zone-content .title {
  font-size: 1em;
}

.file-list li {
  padding: 0.3em 0 0.3em 0.1em;
  height: 2em;
  /* background: yellow; */
}

.file-list li span {
  padding: 0.2em 0;
  color: darkslategrey;
  /* background: blue; */
  height: 1em;
  text-align: left;
}
