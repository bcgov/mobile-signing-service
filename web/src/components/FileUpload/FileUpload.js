//
// DevHub
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2018-09-04.
//

import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './FileUpload.css';

const size = sizeInBytes => {
  return (
    <div>
      &nbsp;&nbsp;
      {Math.round(sizeInBytes * 0.000001)}
      Mb
    </div>
  );
};

const isDisabled = files => {
  if (files.length >= 1) return true;
  return false;
};

const titleForCurrentState = files => {
  if (isDisabled(files)) {
    return <p className="title">Only one file can be selected.</p>;
  }

  return <p className="title">Drag a file to upload (ZIP format only).</p>;
};

const FileUpload = ({ onFileAccepted, files }) => {
  return (
    <div className="file-upload-container">
      <Dropzone className="drop-zone" onDrop={onFileAccepted} disabled={isDisabled(files)}>
        {({ getRootProps, getInputProps }) => (
          <div className="drop-zone-content" {...getRootProps()}>
            <input {...getInputProps()} />
            {titleForCurrentState(files)}
          </div>
        )}
      </Dropzone>
      <ul className="file-list">
        {files.map(file => (
          <li key={file.name}>
            {<FontAwesomeIcon icon="file" className="file-icon" />}
            <span className="file-name">{file.name}</span>
            <span className="file-size">{size(file.size)}</span>
          </li>
        ))}
      </ul>
    </div>
  );
};

FileUpload.propTypes = {
  files: PropTypes.array.isRequired,
  onFileAccepted: PropTypes.func.isRequired,
};

export default FileUpload;
