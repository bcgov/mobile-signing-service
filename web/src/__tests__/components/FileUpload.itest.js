import React from 'react';
import { shallow } from 'enzyme';
import FileUpload from '../../components/FileUpload/FileUpload';
import Dropzone from 'react-dropzone';

describe('FileUpload Component', () => {
  const onFileAccepted = jest.fn();
  const files = [{ name: 'file1', size: 100 }];
  // const files2 = [{ name: 'file1', size: 100 }, { name: 'file2', size: 200 }];

  it('matches snapshot', () => {
    const wrapper = shallow(<FileUpload onFileAccepted={onFileAccepted} files={files} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('enabled by default', () => {
    const wrapper = shallow(<FileUpload onFileAccepted={onFileAccepted} files={[]} />);
    expect(wrapper.find('.drop-zone').prop('disabled')).toBe(false);
  });

  it('disables after a file is added', () => {
    const wrapper = shallow(<FileUpload onFileAccepted={onFileAccepted} files={files} />);
    expect(wrapper.find('.drop-zone').prop('disabled')).toBe(true);
  });

  it('initial text is displayed', () => {
    const wrapper = shallow(<FileUpload onFileAccepted={onFileAccepted} files={files} />);
    expect(
      wrapper
        .find(Dropzone)
        .dive()
        .find('.title')
        .text()
    ).toBe('Only one file can be selected.');
  });
});
