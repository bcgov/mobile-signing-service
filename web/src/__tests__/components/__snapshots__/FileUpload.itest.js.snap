// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload Component matches snapshot 1`] = `
<div
  className="file-upload-container"
>
  <Dropzone
    className="drop-zone"
    disabled={true}
    onDrop={[MockFunction]}
  >
    <Component />
  </Dropzone>
  <ul
    className="file-list"
  >
    <li
      key="file1"
    >
      <FontAwesomeIcon
        border={false}
        className="file-icon"
        fixedWidth={false}
        flip={null}
        icon="file"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span
        className="file-name"
      >
        file1
      </span>
      <span
        className="file-size"
      >
        <div>
            
          0
          Mb
        </div>
      </span>
    </li>
  </ul>
</div>
`;
